client.go

```diff
@@ -924,6 +924,49 @@ func (c *Client) MkdirAll(path string) error {
 	return nil
 }
 
+// RemoveAll delete files recursively in the directory and Recursively delete subdirectories.
+// An error will be returned if no file or directory with the specified path exists
+func (c *Client) RemoveAll(path string) error {
+
+	// Get the file/directory information
+	fi, err := c.Stat(path)
+	if err != nil {
+		return err
+	}
+
+	if fi.IsDir() {
+		// Delete files recursively in the directory
+		files, err := c.ReadDir(path)
+		if err != nil {
+			return err
+		}
+
+		for _, file := range files {
+			if file.IsDir() {
+				// Recursively delete subdirectories
+				err = c.RemoveAll(path + "/" + file.Name())
+				if err != nil {
+					return err
+				}
+			} else {
+				// Delete individual files
+				err = c.Remove(path + "/" + file.Name())
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		// Delete the empty directory
+		return c.RemoveDirectory(path)
+
+	} else {
+		// Delete individual files
+		return c.Remove(path)
+	}
```

Comment:
As a matter of style in Go, if an `if` block ends in a return, then you shouldn’t use an `else` block, since the only way execution could possibly reach here is if the `if` block were never executed.

Reply:
Right, Got it!

Reply:
Hey @puellanivis, any other thoughts, I am modifying this. So that we can close on this?

Reply:
Well, once you move the `return c.Remove(path)` to outside of the `else`, the `return c.RemoveDirectory(path)` will be equivalent to the `return c.Remove(path)`. So, it could just be removed.

Or, in the alternative, since we’re here in the same package, you could switch to the unexported `return c.removeFile(path)`.

Reply:
Please check